### PR TITLE
completed array-manipulation-1 | Janelle Gatmaitan

### PR DESCRIPTION
**Pull request by JanelleGatmaitan**
image
image

### CODE REVIEW

File: array-manipulation-1/src/reverse.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* exported reverse */
+debugger;
```
Comment (thebearingedge) [inline array-manipulation-1/src/reverse.js]: Please remove all `debugger` statements before turning in your solution.

File: array-manipulation-1/src/reverse.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+function reverse(array) {
+  var newArray = [];
+  for (var i = array.length - 1; i >= 0; i--) {
+    newArray = newArray.concat(array[i]);
+  }
+  return newArray;
+}
+reverse([1, 2, 3]);
```
Comment (thebearingedge) [inline array-manipulation-1/src/reverse.js]: Please remove your test code before turning in your solution.

File: array-manipulation-1/src/compact.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/* exported compact */
+debugger;
```
Comment (thebearingedge) [inline array-manipulation-1/src/compact.js]: Please remove all `debugger` statements before turning in your solution.

File: array-manipulation-1/src/compact.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+function compact(array) {
+  var newArray = [];
+  for (var i = 0; i < array.length; i++) {
+    if (array[i]) {
+      newArray.push(array[i]);
+    }
+  }
+  return newArray;
+}
+compact([[], [], []]);
```
Comment (thebearingedge) [inline array-manipulation-1/src/compact.js]: Please remove your test code before turning in your solution.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**